src/ride/track.c

```diff
@@ -646,9 +646,14 @@ rct_track_td6* load_track_design(const char *path)
 		}
 
 		memcpy(&track_design->vehicle_object, vehicle_object, sizeof(rct_object_entry));
-		for (i = 0; i < 32; i++)
+		for (i = 0; i < 32; i++) {
 			track_design->vehicle_additional_colour[i] = track_design->vehicle_colours[i].trim_colour;
 
+			// RCT1 river rapids always had black seats.
+			if (track_design->type == RCT1_RIDE_TYPE_RIVER_RAPIDS)
```

Comment:
Oh, come on! You would have known this if you had looked at the context: this is import code for TD4 designs, which are from RCT1. RCT1 had no adjustable trim colour for river rapids, instead the trim was always coloured black.